buildconfig/CMake/PythonPackageTargetFunctions.cmake

```diff
@@ -59,67 +59,11 @@ function(add_python_package pkg_name)
     DEPENDS ${_setup_py}
   )
 
-  # Generate a sitecustomize.py file in the egg link directory as setuptools no longer generates site.py for v>=49.0.0
-  if(_parsed_arg_GENERATE_SITECUSTOMIZE AND Python_SETUPTOOLS_VERSION VERSION_GREATER_EQUAL 49.0.0)
-    add_custom_command(
-      OUTPUT ${_egg_link_dir}/sitecustomize.py
-      COMMAND ${CMAKE_COMMAND} -DSITECUSTOMIZE_DIR=${_egg_link_dir} -P ${CMAKE_MODULE_PATH}/WriteSiteCustomize.cmake
-      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-      DEPENDS ${_setup_py} ${CMAKE_MODULE_PATH}/WriteSiteCustomize.cmake
-    )
-    list(APPEND _outputs ${_egg_link_dir}/sitecustomize.py)
-  endif()
-
   add_custom_target(${pkg_name} ALL DEPENDS ${_outputs})
 
-  # setuptools by default wants to build into a directory called 'build' relative the to the working directory. We have
-  # overridden commands in setup.py.in to force the build directory to take place out of source. The install directory
-  # is specified here and then --install-scripts=bin --install-lib=lib removes any of the platform/distribution specific
-  # install directories so we can have a flat structure
-  if(CONDA_BUILD)
-    install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E env MANTID_VERSION_STR=${_version_str} \
-    ${Python_EXECUTABLE} -m pip install ${CMAKE_CURRENT_SOURCE_DIR} --no-deps --ignore-installed --no-cache-dir -vvv)"
-            COMPONENT Runtime
-    )
-  else()
-    install(
-      CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E env MANTID_VERSION_STR=${_version_str} \
-    ${Python_EXECUTABLE} ${_setup_py} install -O1 --single-version-externally-managed \
-    --root=${_setup_py_build_root}/install --install-scripts=bin --install-lib=lib \
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})"
-      COMPONENT Runtime
-    )
-  endif()
-
-  if(NOT CONDA_BUILD)
-    # Registers the "installed" components with CMake so it will carry them over
-    if(_parsed_arg_EXCLUDE_FROM_INSTALL)
-      foreach(_dest ${_parsed_arg_INSTALL_LIB_DIRS})
-        install(
-          DIRECTORY ${_setup_py_build_root}/install/lib/
-          DESTINATION ${_dest}
-          COMPONENT Runtime
-          PATTERN "test" EXCLUDE
-          REGEX "${_parsed_arg_EXCLUDE_FROM_INSTALL}" EXCLUDE
-        )
-      endforeach()
-    else()
-      foreach(_dest ${_parsed_arg_INSTALL_LIB_DIRS})
-        install(
-          DIRECTORY ${_setup_py_build_root}/install/lib/
-          DESTINATION ${_dest}
+  install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E env MANTID_VERSION_STR=${_version_str} \
+  ${Python_EXECUTABLE} -m pip install ${CMAKE_CURRENT_SOURCE_DIR} --no-deps --ignore-installed --no-cache-dir -vvv)"
```

Comment:
This seems like quite a complex command, could you add a comment to explain what its doing?

Reply:
Added a comment.